### ISSUE TITLE
Docs: Add a searchbar

### ISSUE THREAD

**Issue by pBouillon** [state-adapt/state-adapt#57]
First of all, thank you for your work! State management always seemed clunky and a burden to setup, with a lot of boilerplate.
With StateAdapt, everything is just straightforward and meaningful at every step, I hope it will still gain traction from the community, and I'm looking forward to [the v2](https://github.com/state-adapt/state-adapt/issues/45) 💪 

StateAdapt features a lot of API and I often find myself looking for the documentation page for a specific method.
Unfortunately, I feel like having to dig into multiple pages to find where it is defined.

I would have like to be able to use a search bar, the same way other libraries such as [NgRx](https://ngrx.io) or [NGXS](https://ngxs.io) do.

This could show results based on the scope (`core`, `angular`, `react`, ...) and the relevant documentation sections, improving both the discoverability of StateAdapt's API and the onboarding of newcommers.

For illustration purposes, here is what NgRx's serachbar looks like when searching for `compo`:

![image](https://github.com/state-adapt/state-adapt/assets/22640284/418ae81d-3bd5-414d-9fef-82ee25a9a0da)

Is this something that you might want to consider?

**Comment by mfp22** [comment]
Yeah, I'll have to look into this. Thanks for bringing it up. I want to focus on the developer experience of using StateAdapt. 

Have you hovered over a method before and looked at the JSDoc tooltip? It has the exact same content as what's on the docs site.

**Comment by mfp22** [comment]
Finally done with new docs site

This particular feature was a piece of cake to add, so the real wait for this was me deciding to completely redo the entire docs site with something more standard. 

<img width="1255" height="823" alt="Image" src="https://github.com/user-attachments/assets/6357f397-caa9-4984-ae02-c0ae3b292655" />
